node/src/BaseClient.ts

```diff
@@ -1144,6 +1147,66 @@ export class BaseClient {
         );
     }
 
+    /**
+     * Reads or modifies the array of bits representing the string that is held at `key` based on the specified
+     * `subcommands`.
+     *
+     * See https://valkey.io/commands/bitfield/ for more details.
+     *
+     * @param key - The key of the string.
+     * @param subcommands - The subcommands to be performed on the binary value of the string at `key`, which could be
+     *      any of the following:
+     *          - {@link BitFieldGet}
+     *          - {@link BitFieldSet}
+     *          - {@link BitFieldIncrBy}
+     *          - {@link BitFieldOverflow}
+     * @returns An array of results from the executed subcommands:
+     *      - {@link BitFieldGet} returns the value in {@link Offset} or {@link OffsetMultiplier}.
+     *      - {@link BitFieldSet} returns the old value in {@link Offset} or {@link OffsetMultiplier}.
+     *      - {@link BitFieldIncrBy} returns the new value in {@link Offset} or {@link OffsetMultiplier}.
+     *      - {@link BitFieldOverflow} determines the behavior of the {@link BitFieldSet} and {@link BitFieldIncrBy}
+     *        subcommands when an overflow or underflow occurs. {@link BitFieldOverflow} does not return a value and
+     *        does not contribute a value to the array response.
+     *
+     * @example
+     * ```typescript
+     * await client.set("key", "A");  // "A" has binary value 01000001
+     * const result = await client.bitfield("key", [new BitFieldSet(UnsignedEncoding(2), new Offset(1), 3), new BitFieldGet(new UnsignedEncoding(2), new Offset(1))]);
+     * console.log(result); // Output: [2, 3] - The old value at offset 1 with an unsigned encoding of 2 was 2. The new value at offset 1 with an unsigned encoding of 2 is 3.
+     * ```
+     */
+    public async bitfield(
+        key: string,
+        subcommands: BitFieldSubCommands[],
+    ): Promise<(number | null)[]> {
+        return this.createWritePromise(createBitField(key, subcommands));
+    }
+
+    /**
+     * Reads the array of bits representing the string that is held at `key` based on the specified `subcommands`.
+     *
+     * See https://valkey.io/commands/bitfield_ro/ for more details.
+     *
+     * @param key - The key of the string.
+     * @param subcommands - The {@link BitFieldGet} subcommands to be performed.
+     * @returns An array of results from the {@link BitFieldGet} subcommands.
+     *
+     * since Valkey version 6.0.0.
+     *
+     * @example
+     * ```typescript
+     * await client.set("key", "A");  // "A" has binary value 01000001
+     * const result = await client.bitfieldReadOnly("key", [new BitFieldGet(new UnsignedEncoding(2), new Offset(1))]);
+     * console.log(result); // Output: [2] - The value at offset 1 with an unsigned encoding of 2 is 2.
+     * ```
+     */
+    public async bitfieldReadOnly(
+        key: string,
+        subcommands: BitFieldGet[],
+    ): Promise<number[]> {
```

Comment:
can the return be null like `bitfield()`? Maybe if the key holds an empty string or a key that is empty?

Reply:
as far as I can tell no, in both cases the command will act as if the key holds a value of all 0s. I think BitFieldSet only returns null if BitOverflowControl is set to FAIL and an overflow occurs.